src/test/java/com/hotels/bdp/circustrain/bigquery/extraction/BigQueryDataExtractionServiceTest.java

```diff
@@ -95,4 +98,26 @@ public void cleanupWhenDeletionFailsDoesntThrowException() {
     verify(storage).delete(any(BlobId.class));
     verify(bucket).delete();
   }
+
+  @Test(expected = CircusTrainException.class)
+  public void jobNoLongerExists() throws InterruptedException {
+    BigQueryExtractionData data = new BigQueryExtractionData(table);
+    TableId tableId = TableId.of("dataset", "table");
+    when(table.getTableId()).thenReturn(tableId);
+    when(table.extract(anyString(), anyString())).thenReturn(job);
+    when(job.waitFor(Matchers.<RetryOption> anyVararg())).thenReturn(null);
+    service.extract(data);
```

Comment:
No big deal but perhaps assert the exception in these tests with ExpectedException, to catch the logic a bit better.  Happy to approve either way

Reply:
Yes, I wasn't happy with not checking the exception message as it could mean the exception being thrown by a different code branch and thus the test not testing what one thinks it does. I see ExpectedException allows you to do to fairly loose "message contains text" checks which I've now used.